### PR TITLE
Shannon/user model

### LINKED ISSUE #4: Create home, help, and about page
this should be under staticpages

### LINKED ISSUE #29: Finish sign-up page with correct forms

### LINKED ISSUE #30: Finish log-in page with right forms

### LINKED ISSUE #31: Create seed data
Make sure al your forms work correctly as well

### LINKED ISSUE #32: On user creation completion, auto-create user-profile page

### PR DESCRIPTION
**Pull request by shannonchu**
Fix #4 
Fix #29
Fix #30
Fix #31 
Fix #32 
Edit nav-bar to have user-profile page link
Create edit user profile button and page

### CODE REVIEW

File: config/initializers/devise.rb [hunk 1]
```diff
@@ -1,6 +1,10 @@
 # Use this hook to configure devise mailer, warden hooks and so forth.
 # Many of these configuration options can be set straight in your model.
 Devise.setup do |config|
+  config.scoped_views = true
+
+  config.secret_key = 'f77c0a9130b11afb017f65ed5a3d027bd54bed53a917902a890594d300cc6d7d89baf3ba209d6cdd53ee84c946a6be686d8f19c28eeb153b8a7db5a79d96d1d0'
```
Comment (hidesminimally) [inline config/initializers/devise.rb]: @shannonchu NOOoOOOOoOOOOoOOOOoOOOOoOOOOoOOOOoOOOOoOOOOoOOOOoOOOOoOOOOoOOOOoOOO WHAT IS THIS NONONONNON  WHY IS THIS HERE 

You should merge your branch with master to get the most recent updates. You'll probably have merge conflicts on the `config.reconfirmable` but remember to comment that line out

File: app/models/user.rb [hunk 2]
```diff
@@ -2,10 +2,15 @@ class User < ActiveRecord::Base
   # Include default devise modules. Others available are:
   # :confirmable, :lockable, :timeoutable and :omniauthable
   devise :database_authenticatable, :registerable,
-         :recoverable, :rememberable, :trackable, :validatable
+         :recoverable, :rememberable, :trackable, :validatable, :confirmable
```
Comment (hidesminimally) [inline app/models/user.rb]: @shannonchu Shouldn't you not need the `:confirmable` module anymore since we've set it to False?

File: test/controllers/user_controller_test.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+require 'test_helper'
```
Comment (hidesminimally) [inline test/controllers/user_controller_test.rb:1]: @shannonchu remove dese useles tests brah

File: test/controllers/static_pages_controller_test.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
+require 'test_helper'
+
+class StaticPagesControllerTest < ActionController::TestCase
+  test "should get home" do
+    get :home
+    assert_response :success
+  end
+
+  test "should get help" do
```
Comment (hidesminimally) [inline test/controllers/static_pages_controller_test.rb:9]: @shannonchu remove dese useles tests TOO BRAH

File: config/routes.rb [hunk 5]
```diff
@@ -1,5 +1,16 @@
 Rails.application.routes.draw do
-  devise_for :users
+
+  root 'static_pages#home'
+
+  devise_for :users, :path => '', :path_names => {:sign_up => 'register', :sign_in => 'login', :sign_out => 'logout'}
+  # devise_for :users, :controllers => { :registrations => "users/registrations_controller" }
```
Comment (hidesminimally) [inline config/routes.rb]: @shannonchu  remove dem comments

**Comment by hidesminimally** [comment]
@shannonchu Get https://github.com/jish/pre-commit! It'll help with your styling. Also, in Sublime, under View>Indentation, make sure tab width is two and indent using spaces is checked.

File: app/controllers/user_controller.rb [hunk 6]
```diff
@@ -0,0 +1,22 @@
+class UserController < ApplicationController
+  def show
+  	@user = User.find(params[:id])
+  	@first_name = @user.first_name
```
Comment (hidesminimally) [inline app/controllers/user_controller.rb]: @shannonchu I don't think you need to make these extra variables

File: test/controllers/user_controller_test.rb (hunk 3 above)
Comment (shannonchu) [inline test/controllers/user_controller_test.rb:1]: @lilshim i dont have this file

File: test/controllers/static_pages_controller_test.rb (hunk 4 above)
Comment (shannonchu) [inline test/controllers/static_pages_controller_test.rb:9]: @lilshim I dont have this file either

File: app/views/layouts/_messages.html.erb [hunk 7]
```diff
@@ -0,0 +1,5 @@
+<% flash.each do |name, msg| %>
+  <% if msg.is_a?(String) %>
```
Comment (hidesminimally) [inline app/views/layouts/_messages.html.erb:2]: @shannonchu You needed this in your layout to render the `flash[:notice]`

File: app/views/users/registrations/edit.html.erb [hunk 8]
```diff
@@ -28,29 +28,6 @@
     <%= f.file_field :photo %>
   </div>
```
Comment (hidesminimally) [inline app/views/users/registrations/edit.html.erb:30]: @shannonchu deleted this because it doesn't belong in edit-profile page. should make an edit account settings button for this

**Comment by hidesminimally** [comment]
@shannonchu Take a look at the changes I've made. Shoot me a message if anything is unclear!
